### ISSUE TITLE
(run prog ...) with prog an installable program does not work on Windows

### ISSUE THREAD

**Issue by nojb** [ocaml/dune#1263]
To reproduce, consider
```
(rule (with-stdout-to foo.ml (echo "")))
(executable (name foo) (public_name foo.exe))
(alias (name bar) (action (run foo.exe)))
``` 
Then `dune build @bar` fails with
```
Error: Error: Program foo.exe not found in the tree or in PATH (context: default)
```

**Comment by diml** [comment]
does it work if you remove the .exe extension? Dune will add it automatically on Windows.

**Comment by nojb** [comment]
Yes, it works. OK, is the idea that I never write the `.exe` extension and it will be added automatically on Windows ? (also for `public_name`?)

Thanks!

**Comment by diml** [comment]
Yep, that's the idea. It's seemed like the simplest solution to improve portability when we looked at this with @dra27

**Comment by nojb** [comment]
The only downside is the asymmetry with local (non-installed) programs, where you need to use `%{ext_exe}`, it seems.

**Comment by diml** [comment]
Dune systematically uses the `.exe` extension for executables, so for local programs you can always write `(run ./foo.exe ...)`. On Unix, it removes it when installing executables.

I guess we could support `(run ./foo ...)` as well.
